clippy_lints/src/strings.rs

```diff
@@ -169,3 +169,61 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for StringLitAsBytes {
         }
     }
 }
+
+/// **What it does:** Checks for usage of `.repeat(1)` since its equivalent to clone()
+///
+/// **Why is this bad?** `.repeat(1)` is not very readable.
+///
+/// **Example:**
+/// ```rust
+/// fn main() {
+///     let x = String::from("hello world").repeat(1);
+/// }
+/// ```
+/// This is the same as:
+/// 
+/// ```rust
+/// fn main() {
+///     let x = String::from("hello world").clone();
+/// }
+/// ```
+declare_clippy_lint! {
+    pub STRING_REPEAT_ONCE,
+    pedantic,
```

Comment:
I would classify this as a complexity lint. If too many people complain about this lint (which I don't think) we can move this to pedantic later.